lib/ReadsAlignmentUtils/ReadsAlignmentUtilsImpl.py

```diff
@@ -242,92 +208,108 @@ def _get_aligner_stats(self, bam_file):
-
-                if flag[-2] == '1':
-                    properly_paired += 1
+                    mapped_single_end_reads_ids.append(reads_id)
+                    if alignment.is_secondary:
+                        secondary_alignment_count += 1
+                        secondary_alignment_single_end_reads_ids.append(reads_id)
 
         infile.close()
 
-        if paired:
-            mapped_reads_ids = mapped_left_reads_ids + mapped_right_reads_ids
-            unmapped_reads_count = unmapped_left_reads_count + unmapped_right_reads_count
-
-            mapped_reads_ids_counter = Counter(mapped_reads_ids)
-            mapped_reads_count = len(list(mapped_reads_ids_counter))
 
-            singletons = list(mapped_reads_ids_counter.values()).count(1)
-            multiple_alignments = mapped_reads_count - singletons
+        #prepare  summary
+        mapped_left_reads_count = len(set(mapped_left_reads_ids))
+        mapped_right_reads_count = len(set(mapped_right_reads_ids))
+        both_pair_mapcount = len(set(mapped_left_reads_ids) & set(mapped_right_reads_ids))
+        singletons = mapped_left_reads_count + mapped_right_reads_count - both_pair_mapcount*2
 
-            total_reads = unmapped_reads_count + mapped_reads_count
+        mapped_reads_count = mapped_left_reads_count + \
+                                 mapped_right_reads_count + \
+                                 len(set(mapped_single_end_reads_ids))
 
-            properly_paired = properly_paired // 2
-
-        else:
-            mapped_reads_ids_counter = Counter(mapped_reads_ids)
-            mapped_reads_count = len(list(mapped_reads_ids_counter))
 
-            singletons = list(mapped_reads_ids_counter.values()).count(1)
-            multiple_alignments = mapped_reads_count - singletons
+        total_reads_count = mapped_reads_count + unmapped_reads_count
 
-            total_reads = unmapped_reads_count + mapped_reads_count
+        #count for reads that are aligned in multiple places
+        multiple_alignments = len(set(secondary_alignment_left_reads_ids)) + \
+                              len(set(secondary_alignment_right_reads_ids)) + \
+                              len(set(secondary_alignment_single_end_reads_ids))
 
         try:
-            alignment_rate = round(float(mapped_reads_count) / total_reads * 100, 3)
+            alignment_rate = round(float(mapped_reads_count) / total_reads_count * 100, 3)
         except ZeroDivisionError:
             alignment_rate = 0
 
-        if alignment_rate > 100:
-                alignment_rate = 100.0
-
         elapsed_time = time.time() - start_time
         self.__LOGGER.info('Used: {}'.format(time.strftime("%H:%M:%S", time.gmtime(elapsed_time))))
 
+
         stats_data = {
             "alignment_rate": alignment_rate,
             "mapped_reads": mapped_reads_count,
             "multiple_alignments": multiple_alignments,
-            "properly_paired": properly_paired,
```

Comment:
should `properly_paired` be kept in return?

Reply:
fixed